reactor-netty-examples/src/main/java/reactor/netty/examples/http/snoop/HttpSnoopServer.java

```diff
@@ -0,0 +1,166 @@
+		// 1.create and config the server instance
+		HttpServer server = HttpServer.create()
+		                              .port(PORT)
+		                              .wiretap(WIRETAP)
+		                              .compress(COMPRESS);
+
+		if (SECURE) {
+			SelfSignedCertificate ssc = new SelfSignedCertificate();
+			if (HTTP2) {
+				server = server.secure(spec -> spec.sslContext(Http2SslContextSpec.forServer(ssc.certificate(), ssc.privateKey())));
+			}
+			else {
+				server = server.secure(spec -> spec.sslContext(Http11SslContextSpec.forServer(ssc.certificate(), ssc.privateKey())));
+			}
+		}
+		if (HTTP2) {
+			server = server.protocol(HttpProtocol.H2);
+		}
+
+		// 2.config the route rule, this server will receive any request that has any path and any method,
+		// and then send back the details of the received HTTP request
+		server = server.route(routes -> routes.route(
+				req -> true,
+				HttpSnoopServer::parseRequestAndSendResponse
+		));
+
+		// 3. start the server and block the main thread to keep the server running
+		server.bindNow().onDispose().block();
+	}
+
+	private static NettyOutbound parseRequestAndSendResponse(HttpServerRequest request, HttpServerResponse response) {
+		StringBuilder buf = new StringBuilder();
+		buf.append("request received, the details are: \n\n");
+
+		// append the request method, uri and protocol version
+		buf.append("your host: ").append(request.hostAddress()).append("\n");
+		buf.append("http method: ").append(request.method().name()).append("\n");
+		buf.append("http version: ").append(request.version().text()).append("\n");
+		buf.append("request path: ").append(request.path()).append("\n\n");
+
+		// append the request headers
+		buf.append("headers: \n");
+		request.requestHeaders().forEach(entry -> {
+			buf.append(entry.getKey());
+			buf.append(" : ");
+			buf.append(entry.getValue());
+			buf.append("\n");
+		});
+		buf.append("\n");
+
+		// append the query parameters
+		buf.append("query parameters: \n");
+		new QueryStringDecoder(request.uri()).parameters().forEach((key, value) -> {
+			buf.append(key);
+			buf.append(" : ");
+			buf.append(value);
+			buf.append("\n");
+		});
+		buf.append("\n");
+
+		// append the request content if there is any
+		buf.append("content: \n");
+		String contentType = request.requestHeaders().get(HttpHeaderNames.CONTENT_TYPE);
+		Mono<String> responseContent;
+		if (isATextualContentType(contentType)) {
+			responseContent = request.receive().aggregate().asString()
+				   .onErrorResume(e -> {
+					   System.out.println("error occurs when receiving the content: " + e.getMessage());
+					   return Mono.empty();
+				   })
+				   .switchIfEmpty(Mono.just(""))
+			       .map(content -> {
+						buf.append(content);
+						buf.append("\n");
+						return buf.toString();
+					});
+		} else {
+			buf.append("there is no content or the content is not textual, so we don't print it here\n");
+			responseContent = Mono.just(buf.toString());
+		}
```

Comment:
```suggestion
		} 
		else {
			buf.append("there is no content or the content is not textual, so we don't print it here\n");
			responseContent = Mono.just(buf.toString());
		}
```

minor: please add a new line before the `else`.